app/models/ability.rb

```diff
@@ -0,0 +1,35 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    #
+    can :read, :all
+    if user.admin?
+      can :manage, :all
+    else
+      can :create, Comment
+      can :manage, Comment, user: user
+      can :manage, Post, user:
+    end
```

Comment:
- [ ] Kindly make sure you add this: `can :read, :all` inside the else condition to make sure the user can read all the posts and comments 
- [ ] Make sure to use `author` instead of the `user` since the Post and Comment has `author_id`, as the foreign key to reference user table, your code should looks like this:

```ruby
     can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
     can :manage, Comment, author: user
     can :read, :all
```